ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureClient.java

```diff
@@ -144,45 +144,55 @@ public void onResult(@Nullable Configuration configuration, @Nullable Exception
             @Override
-            public void onResult(@Nullable Configuration configuration, @Nullable Exception error) {
-                if (configuration == null) {
-                    callback.onResult(null, null, error);
-                    return;
+            public void onAuthorizationResult(@Nullable Authorization authorization, @Nullable Exception authError) {
+                if (authorization != null) {
 
-                }
-                if (configuration.getCardinalAuthenticationJwt() == null) {
-                    Exception authError = new BraintreeException("Merchant is not configured for 3DS 2.0. " +
-                            "Please contact Braintree Support for assistance.");
-                    callback.onResult(null, null, authError);
-                    return;
-                }
+                    final JSONObject lookupJSON = new JSONObject();
+                    try {
+                        lookupJSON
+                                // TODO: call async getAuthorization here
+                                .put("authorizationFingerprint", authorization.getBearer())
+                                .put("braintreeLibraryVersion", "Android-" + BuildConfig.VERSION_NAME)
+                                .put("nonce", request.getNonce())
+                                .put("clientMetadata", new JSONObject()
+                                        .put("requestedThreeDSecureVersion", "2")
+                                        .put("sdkVersion", "Android/" + BuildConfig.VERSION_NAME));
+                    } catch (JSONException ignored) {
+                    }
 
-                cardinalClient.initialize(context, configuration, request, new CardinalInitializeCallback() {
-                    @Override
-                    public void onResult(String consumerSessionId, Exception error) {
-                        if (consumerSessionId != null) {
-                            try {
-                                lookupJSON.put("dfReferenceId", consumerSessionId);
-                            } catch (JSONException ignored) {
+                    braintreeClient.getConfiguration(new ConfigurationCallback() {
+                        @Override
+                        public void onResult(@Nullable Configuration configuration, @Nullable Exception configError) {
+                            if (configuration == null) {
+                                callback.onResult(null, null, configError);
+                                return;
+                            }
+                            if (configuration.getCardinalAuthenticationJwt() == null) {
+                                Exception authError = new BraintreeException("Merchant is not configured for 3DS 2.0. " +
+                                        "Please contact Braintree Support for assistance.");
+                                callback.onResult(null, null, authError);
```

Comment:
```suggestion
                                Exception configError = new BraintreeException("Merchant is not configured for 3DS 2.0. " +
                                        "Please contact Braintree Support for assistance.");
                                callback.onResult(null, null, configError);
```

Reply:
This is technically still an `authError` with cardinal. This block of code didn't change in this PR it just got moved.